qa/tasks/keystone.py

```diff
@@ -104,9 +120,12 @@ def install_packages(ctx, config):
     assert isinstance(config, dict)
     log.info('Installing packages for Keystone...')
 
+    patch_bindep = patch_bindep_template \
+        .replace("{keystone_dir}", get_keystone_dir(ctx))
     packages = {}
     for (client, _) in config.items():
         (remote,) = ctx.cluster.only(client).remotes.keys()
+        toxvenv_sh(ctx, remote, ['python'], stdin=patch_bindep)
```

Comment:
there's a similar hack above for requirements.txt that just uses sed:
```python
        # hax for http://tracker.ceph.com/issues/23659
        run_in_keystone_dir(ctx, client, [
                'sed', '-i',
                's/pysaml2<4.0.3,>=2.4.0/pysaml2>=4.5.0/',
                'requirements.txt'
            ],
        )
```
think its worth changing it?

Reply:
I thought about it, but decided it wasn't worth the bother.  Also with python I can make it emit a slightly more informative  message in the logs if it can't find the string.

Reply:
I would prefer 'sed' way. Also, when you ask for 'python' on some systems it means 'python2', on another 'python3', so you should be very careful about your python snippet.